app/client/src/components/pages/Community/components/tabs/Overview.js

```diff
@@ -35,334 +42,839 @@ import {
+  // and draw them on the map
+  useEffect(() => {
+    if (services.status === 'fetching') return;
+    if (!monitoringStations.data.features) return;
+
+    const stations = monitoringStations.data.features.map((station) => ({
+      sampleType: 'Monitoring Station',
+      siteId: station.properties.MonitoringLocationIdentifier,
+      orgId: station.properties.OrganizationIdentifier,
+      orgName: station.properties.OrganizationFormalName,
+      locationLongitude: station.geometry.coordinates[0],
+      locationLatitude: station.geometry.coordinates[1],
+      locationName: station.properties.MonitoringLocationName,
+      locationType: station.properties.MonitoringLocationTypeName,
+      // TODO: explore if the built up locationUrl below is ever different from
+      // `station.properties.siteUrl`. from a quick test, they seem the same
+      locationUrl:
+        `${services.data.waterQualityPortal.monitoringLocationDetails}` +
+        `${station.properties.ProviderName}/` +
+        `${station.properties.OrganizationIdentifier}/` +
+        `${station.properties.MonitoringLocationIdentifier}/`,
+      // monitoring station specific properties:
+      stationProviderName: station.properties.ProviderName,
+      stationTotalSamples: station.properties.activityCount,
+      stationTotalMeasurements: station.properties.resultCount,
+    }));
+
+    setNormalizedMonitoringStations(stations);
+
+    plotStations(Graphic, stations, monitoringStationsLayer, services);
+  }, [Graphic, monitoringStations.data, monitoringStationsLayer, services]);
+
+  const allSampleLocations = [
+    ...normalizedUsgsStreamgages,
+    ...normalizedMonitoringStations,
+  ];
+
+  const [sampleLocationsSortedBy, setSampleLocationsSortedBy] = useState(
+    'locationName',
+  );
+
+  const sortedSampleLocations = [...allSampleLocations].sort((a, b) => {
+    if (sampleLocationsSortedBy === 'stationTotalMeasurements') {
+      return (
+        (b.stationTotalMeasurements || 0) - (a.stationTotalMeasurements || 0)
+      );
+    }
+
+    if (sampleLocationsSortedBy === 'siteId') {
+      return a.siteId.localeCompare(b.siteId);
+    }
+
+    return a[sampleLocationsSortedBy].localeCompare(b[sampleLocationsSortedBy]);
+  });
+
+  const filteredSampleLocations = sortedSampleLocations.filter((item) => {
+    const displayedTypes = [];
+    if (usgsStreamgagesDisplayed) displayedTypes.push('USGS Streamgage');
+    if (monitoringStationsDisplayed) displayedTypes.push('Monitoring Station');
+    return displayedTypes.includes(item.sampleType);
+  });
+
+  if (
+    monitoringStations.status === 'fetching' ||
+    usgsStreamgages.status === 'fetching'
+  ) {
+    return <LoadingSpinner />;
+  }
+
+  if (
+    monitoringStations.status === 'failure' &&
+    usgsStreamgages.status === 'failure'
+  ) {
+    return (
+      <div css={modifiedErrorBoxStyles}>
+        <p>{monitoringError}</p>
+      </div>
+    );
+  }
```

Comment:
How to handle failure cases for when **only one** of the two web services is a good question. I think there's a few ways we could handle it:

1) If both web services have a shared error message **and we don't care if one web service is down** (as we gracefully handle a null response as an empty dataset), we'd check for a failure state on both using a "logical and".

2) If we wanted individual error messages for each web service, we'd want to check for a failure on each, and show individual error messages for each. That would mean updating the code to replace this single if statement with two conditional rendering blocks inside the "success" if statement below (which I'm updating to use a logical OR operator – see my next comment)

Option 1 is how it's currently implemented, so I think we should keep it as is for now, and ask EPA if they want individual error messages for each web service (Option 2), or some other in-between option I'm not considering 😄 

> This could lead to nothing being displayed if one of the services were to fail. Since the code below is looking for both services to return a success, this code should do an or check on failure.

Good catch! I'll update the success checking below to use a logical OR operator when checking for web service successes.